crates/router/src/core/refunds.rs

```diff
@@ -307,7 +305,7 @@ pub async fn refund_retrieve_core(
             merchant_account.storage_scheme,
         )
         .await
-        .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)?;
+        .to_not_found_response(errors::ApiErrorResponse::InternalServerError)?;
```

Comment:
Why is it a `to_not_found_response` but you are throwing internal server back. Should be `change_context(InternalServerError)`

Reply:
At this point if payment_attempt is not found It should be InternalServerError because refund was created and payment was found. 
Also to_not_found_response has something as `CustomerRedacted` . So it is possible that this error is also thrown.

Reply:
Maybe then can you create a seperate trait for it instead of using to_not_found_response

Reply:
It is being used across the code base. Will take this up in a separate PR if required.

Reply:
It's only used for this use case

You can create something like

```rust
trait CustomerRedactError {
	fn to_customer_redact_error();
}

trait ToNotFoundResponse: CustomerRedactError {
	fn to_not_found_response(&self,error: ApiErrorResponse) {
			self.to_customer_redact_error();	
	}
}
```

Reply:
You can create an issue for it.